### PR TITLE
eza: add new package

### PR DESCRIPTION
**Pull request by oskarirauta**
eza is a substitute for ls command with icons and colors and more. Common for those who use ohmyzsh or similar shell stylers.

Maintainer: Jonas Jelonek / @jonasjelonek 
Compile tested: x86_64, latest git
Run tested: x86_64, latest git

### CODE REVIEW

**Comment by oskarirauta** [comment]
Could someone point me out, what is so great with rust? Simple ls command has huge amount of deps and resulting binary takes 1.5mb :O that's 3 times size of whole busybox....

**Comment by jonasjelonek** [comment]
> Could someone point me out, what is so great with rust? Simple ls command has huge amount of deps and resulting binary takes 1.5mb :O that's 3 times size of whole busybox....

I guess there are enough articles explaining what concepts Rust uses to address several issues that e.g. a language like C has, and how Rust could really improve security, safety, etc. for quite some things. So I won't try to explain that here, Google is your friend ;)
I really like Rust, not only because of its features, but also because of its syntax and ecosystem. But that's just my opinion.

But yes, the binary size is still a thing, especially because OpenWrt targets to be rather small. You can do quite some tuning to reduce the size (as explained [here](https://github.com/johnthagen/min-sized-rust)), but `eza` already includes some things in their `Cargo.toml` for that. The remaining issue is, that the binary is statically compiled since Rust's libstd is not available on OpenWrt (yet). In contrast, a C standard library is included in OpenWrt so most packages are dynamically compiled, and thus significantly smaller.
And besides that, `eza` has more features so I would expect it to be bigger anyway.

**Comment by oskarirauta** [comment]
Yes, I've read that all.  Optionals to avoid usage of pointers, which is a direct copy from Swift - which ofcourse is missing a missing system unless you are using Ubuntu, where it also is very limited due to that it is developed and designed by Apple for targeted audience and systems. I don't like the eco system, it's like Boost libraries on c++ - I wouldn't write my own curl library, but for most of stuff, I want to write my own as the advantage of using a centralised library in itself is a huge risk when considering to security, millions of programs become insecure when one of them has a issue- and due to statical linking, program outputting "Hello world" gives a huge binary - optionals are great, but it's just more a lack of coder's skills in remembering to free pointers. In my case it's a poor man's swift except everything written in it is huge and everyone should be using centralised libraries, where in my case, it reminds of language basic - great for beginners and those who can't. These days carbon footprint is a huge thing, and it should be considered in things like these as well, I am not that eco-friendly, but something has to be wrong if in assembler it would take less than a kb - and on something else, even if it's not linked with everything being statical and all, huge capacity of binary in it's self is a risk.

File: utils/eza/Makefile
Comment (BKPepe) [inline utils/eza/Makefile:15]:
```suggestion
PKG_BUILD_DEPENDS:=rust/host
```

small nitpicking, nothing else, sorry!

File: utils/eza/Makefile
Comment (BKPepe) [inline utils/eza/Makefile:4]: My 2c: Hmm, hmm... they are releasing very often. I see there is a new version. This will be very hard to maintain in terms of keeping this up to date. One guy complained recently that so many packages are so old in this repo... Dont take it wrong. Just my thoughts.

**Review by BKPepe — APPROVED** [review]
The Makefile looks great! 👏 👏  As I said in different PR, you learned so much that now I dont have anything else to add. Thank you!

**Comment by oskarirauta** [comment]
**Hold on, at least before merging!**
Some conversation about package is necessary due to facts unknown to me before.

If they push that often, I must admit that I am not going to be able to maintain it that often. It would be wrong for the community to me maintain this if they push new one once in a week; though I wonder, what they could improve with it every week, in the end; this is a replacement for command to list files. I reviewed changelogs for some of last releases for this; mostly they seem to have updated project's submodules (or crates in rust?) - so it doesn't change user experience a lot, so it probably doesn't matter if it won't be updated for every release, but that's my thought only.

Anyway, it's a shame that it's not the other way around; lsd (LS deluxe) that I pushed at same time with this, is very similar to this; but I like eza a lot more, how it displays directory listing with -l argument looks way better, so I would take this over lsd any day; but I do realise that if it is necessary to be updating this every week- it could become a issue, so should we drop this?

**Comment by oskarirauta** [comment]
> The Makefile looks great! 👏 👏 As I said in different PR, you learned so much that now I dont have anything else to add. Thank you!

You are welcome.. I'll clean up that extra space and update this after decision on how to proceed with this due to too often updates in mainstream.

File: utils/eza/Makefile
Comment (oskarirauta) [reply utils/eza/Makefile]: None taken.
Concern about this is in place. I cannot volunteer to update it once in a week if they keep up with that pace.

**Comment by jonasjelonek** [comment]
If nobody minds, I would volunteer to take over ;).   
In my daily usage I mostly replaced ls with eza. Thus, it would be quite nice to have that package in OpenWrt and I wouldn't mind to update it that often.

**Comment by oskarirauta** [comment]
@jonasjelonek 

Sounds great to me! I can add you as the maintainer for the package for you.

**Comment by jonasjelonek** [comment]
Would you mind changing my email to 'jelonek.jonas@gmail.com'? I'm rather using that one ... sorry, my bad. (I should also change that on my github profile ;) ).

otherwise LGTM

**Comment by oskarirauta** [comment]
> Would you mind changing my email to [jelonek.jonas@gmail.com](mailto:jelonek.jonas@gmail.com) ? I'm rather using that one ... sorry, my bad. (I should also change that on my github profile ;) ).
> 
> otherwise LGTM

Done.

**Review by 1715173329 — APPROVED** [review]

**Pull request merged**
